### ISSUE TITLE
Literal of ES2020 Violates the Principle of Unique

### ISSUE THREAD

**Issue by clhiker** [estree/estree#282]
The type of Literal of ES2020 is same as type of ES5, which violates the principle of uniqueness.
I found this problem when I used the uniqueness principle to resolve non-terminators backward compatible

**Comment by JLHwung** [comment]
> The type of Literal of ES2020 is same as type of ES5

Yes, this is how we extend current AST nodes. I don't think it violates the principle of uniqueness. On the contrary, the Literal type is an important example of uniqueness.

> when I used the uniqueness principle to resolve non-terminators backward compatible

Can you elaborate?
